src/lib/swapper/swappers/ThorchainSwapper/utils/addAggregatorAndDestinationToMemo.ts

```diff
@@ -36,6 +38,20 @@ export const addAggregatorAndDestinationToMemo = ({
     'expected finalAssetLimitWithManualSlippage to be a positive amount',
   )
 
+  const minimumPrecision = 5
+  const endingExponential = finalAssetPrecision - minimumPrecision
+
+  // The uniswap aggregator expects this amount to be an exponent, we need to add two numbers at the end which are used at exponents in the contract
+  // We trim 10 of precisions to make sure the THORChain parser can handle the amount without precisions and rounding issues
+  // If the finalAssetPrecision is under 5, the THORChain parser won't fail and we add one exponent at the end so the aggregator contract won't multiply the amount
+  const finalAssetLimitWithTwoLastNumbersAsExponent =
+    finalAssetPrecision <= 5
+      ? `${finalAssetLimitWithManualSlippage}01`
+      : `${bn(finalAssetLimitWithManualSlippage)
+          .dividedBy(10 ** finalAssetPrecision)
+          .toFixed(minimumPrecision, BigNumber.ROUND_DOWN)
+          .replace('.', '')}${endingExponential < 10 ? `0${endingExponential}` : endingExponential}`
```

Comment:
While you have tested this to be working, feels quite hacky - can't we achieve the same output using `toExponent(5)` (where 5 may be programmatic) from the base unit amount instead, and doing a `.replace('e+', '')

Reply:
No as toExponent will output a float and the THORChain parser will keep the integer it will result in a big gap:

https://go.dev/play/p/o5lzkbMpZUp